src/Axe.Windows.Telemetry/Logger.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Axe.Windows.Telemetry
+{
+    /// <summary>
+    /// Wraps telemetry logging and sends it to 
+    /// a supplied IAxeWindowsTelemetry implementation
+    /// </summary>
+    public static class Logger
+    {
+        private static IAxeWindowsTelemetry _telemetry = null;
+        private static readonly Object _lockObject = new Object();
+
+        /// <summary>
+        /// Whether or not telemetry is enabled. Exposed to allow callers who do lots of
+        /// work to short-circuit their processing when telemetry is disabled
+        /// </summary>
+        public static bool IsEnabled => _telemetry != null && _telemetry.IsEnabled;
+
+        /// <summary>
+        /// Sets the telemetry sink for Axe.Windows.
+        /// <param name="telemetry">Telemetry from Axe.Windows will be forwarded to this object</param>
+        public static void SetTelemetrySink(IAxeWindowsTelemetry telemetry)
+        {
+            lock (_lockObject)
+            {
+                _telemetry = telemetry;
+            }
+        }
+
+        /// <summary>
+        /// Publishes event with single property/value pair to the current telemetry pipeline
+        /// </summary>
+        /// <param name="action"></param>
+        /// <param name="property"></param>
+        /// <param name="value"></param>
+        public static void PublishTelemetryEvent(TelemetryAction action, TelemetryProperty property, string value)
+        {
+            // Check IsEnabled because ToString on enums is expensive
+            if (!IsEnabled) return;
+
+            PublishTelemetryEvent(action, new Dictionary<TelemetryProperty, string>
+                    {
+                        { property, value }
+                    });
```

Comment:
Tabs are off here

Reply:
@RobGallo, ideally lines 43-45 would each have one less indent

---
In reply to: [272367311](https://github.com/Microsoft/accessibility-insights-windows/pull/258#discussion_r272367311) [](ancestors = 272367311)